src/ProjectEnvironmentConfiger.ts

```diff
@@ -78,54 +66,48 @@ export class ProjectEnvironmentConfiger {
   static async configureProjectEnvironmentAsPlatform(
       context: vscode.ExtensionContext, channel: vscode.OutputChannel,
       telemetryContext: TelemetryContext, platform: PlatformType,
-      deviceRootPath: string, scaffoldType: ScaffoldType): Promise<boolean> {
+      projectFileRootPath: string,
+      scaffoldType: ScaffoldType): Promise<boolean> {
+    let project;
     if (platform === PlatformType.Arduino) {
-      // First ensure the project is correctly open.
-      const iotProject = await utils.constructAndLoadIoTProject(
-          context, channel, telemetryContext);
-      if (!iotProject) {
-        return false;
-      }
-
-      // Validate platform.
-      // Only iot workbench Arduino project created by workbench extension
-      // can be configured as Arduino platform(for upgrade).
+      // Verify it is an iot workbench Arduino project
       const projectHostType = await IoTWorkbenchProjectBase.getProjectType(
-          scaffoldType, deviceRootPath);
+          scaffoldType, projectFileRootPath);
       if (projectHostType !== ProjectHostType.Workspace) {
         const message =
-            `This is not an iot workbench Arduino projects. You cannot configure it as Arduino platform.`;
+            `This is not an iot workbench Arduino project. You cannot configure it as Arduino platform.`;
         utils.channelShowAndAppendLine(channel, message);
         vscode.window.showWarningMessage(message);
         return false;
       }
-    }
 
-    let project;
-    if (platform === PlatformType.EmbeddedLinux) {
+      project = new ioTWorkspaceProjectModule.IoTWorkspaceProject(
+          context, channel, telemetryContext);
+      if (!project) {
+        // Ensure the project is correctly open.
+        await utils.handleIncorrectlyOpenedIoTWorkspaceProject(
```

Comment:
How about name the function as "ensureProjectIsOpen" to make method name easy to understand?

Reply:
Fixed.